packages/flutter_tools/lib/src/commands/custom_devices.dart

```diff
@@ -443,13 +434,13 @@ class CustomDevicesAddCommand extends CustomDevicesCommandBase {
         // find a random port we can forward
         final int port = await _operatingSystemUtils.findFreePort();
 
-        final ForwardedPort forwardedPort = await (portForwarder.tryForward(port, port) as FutureOr<ForwardedPort>);
+        final ForwardedPort? forwardedPort = await portForwarder.tryForward(port, port);
         if (forwardedPort == null) {
           _printConfigCheckingError("Couldn't forward test port $port from device.",);
           result = false;
         }
 
-        await portForwarder.unforward(forwardedPort);
+        await portForwarder.unforward(forwardedPort!);
```

Comment:
Can you move this to an `else` block, lest we hit a _CastError with the bang operator?

Reply:
Done.